usr/src/cmd/zdb/zdb.c

```diff
@@ -1111,7 +1116,7 @@ dump_history(spa_t *spa)
 	umem_free(buf, SPA_MAXBLOCKSIZE);
 
 	(void) printf("\nHistory:\n");
-	for (int i = 0; i < num; i++) {
+	for (unsigned i = 0; i < num; i++) {
```

Comment:
Do you plan to make this change in the rest of the ZFS codebase?  I think it would be quite extensive.  Can you expand a bit on what the benefit is?

TBH, I'm OK with making the changes to ZDB, but it's going to be hard to keep following these rules without compiler assistance.